content/2017.12.kinto-at-mozilla.rst

```diff
@@ -0,0 +1,119 @@
+
+
+Firefox remote settings
+=======================
+
+The historical one!
+
+Firefox addons and security engineers edit remote settings like blocklist or certificates revocation lists via the Kinto Admin UI. Reviewing and sign-off features are powered by `kinto-signer`. The several hundreds millions of Firefox browsers regularly poll for changes using the `kinto-changes` endpoint. Two instances are deployed with the same database, a public one read-only behind a CDN, and a private one only accessible via a secured VPN.
+
+The client lies in the Firefox codebase, implemented with `kinto.js`. It verifies `content signatures` and has a specific adapter to store local data in SQlite instead of IndexedDB. Some `Telemetry data` is sent when remote settings are synchronized locally in order to track uptake globally.
+
+In order to manage buckets and collections, as well as groups and their related permissions, we use `kinto-wizard` that reads a YAML file from a Github private repo, and creates or updates the missing objects.
+
+Instance: https://firefox.settings.services.mozilla.com
+Production date:
+Average requests/sec: X (*from N CDN*)
+
+Android assets
+--------------
+
+- fonts, etc
+- kinto-attachment
+
+Android and iOS experiments
+---------------------------
+
+This stack is also used to deliver A/B testing features on Android and iOS.
+
+Abandonned use-cases...
+-----------------------
+
+- fingerprint fonts
+
+Possible evolutions
+-------------------
+
+- There is certainly going to be more types of settings for new use-cases
+- The stack currently uses `kinto-ldap` to authenticate users. We'll switch to OpenID Connect now that the company uses Auth0 everywhere.
+- The `Push team` is working on a solution to replace frequent polling with a push (at that scale nothing is trivial).
+- The Sync team is working on a new data store for profile data. We'll probably replace the current SQlite adapter.
+
+
+WebExtensions storage.sync API
+==============================
+
+The tough one!
+
+Firefox is now compatible with the Web Extensions APIs, that allow addons developers to share a common code-base between different browsers. One of these APIs allows addons to store and sync some data (eg. user-preferences).
+In our implementation, the client encrypts the data before sending it to our servers to guarantee privacy. It might not be the case of Chrome by the way. It uses `kinto.js` and leverages the Firefox Accounts keyring.
+
+This stack allowed us to really stress Kinto. When popular extensions like Adblock switched to Web Extensions, we had database congestions and lots of 5XX errors. We should have been a lot more thorough with load testing, and hence discovered race conditions and scaling bugs the hard way. Most of them are fixed now though.
+
+Instance:
+Production date: July 2017
+Average requests/sec: 1100 (70K/min)
```

Comment:
I am not sure this information can be made public unfortunately sames for the number of calls on kinto-settings.